lib/dialects/mysql/query-generator.js

```diff
@@ -149,7 +149,7 @@ class MySQLQueryGenerator extends AbstractQueryGenerator {
   }
 
   showTablesQuery() {
-    return 'SHOW TABLES;';
+    return "SELECT TABLE_NAME FROM INFORMATION_SCHEMA.TABLES WHERE TABLE_TYPE = 'BASE TABLE';";
```

Comment:
This introduced another bug. It shows tables for all schemas not just the current schema.

Reply:
Well, not a bug since it is how the API works in all other dialects. But an API breaking change then.

Reply:
Btw, you mean the default schema for that user ?

Reply:
in mysql databases are schemas so when you select a database in sequelize you are actually selecting a schema but this query returns tables for all databases/schemas not just the selected database/schema.

Reply:
The correct query would probably be `show full tables where Table_Type = 'BASE TABLE'`

Reply:
How, so `SHOW TABLES` actually works like a SELECT statement ? I do not have a mysql db locally to test so I relied on the CI jobs. You can make a small PR just to revert that change. 
But when #11451 it taken care of, showAllTables will be normalized across all dialects and that means that all schemas are returned. But that will be done during a major release so API breakage is expected.

Reply:
Hmmm I see now that it's a bit more complicated than I thought, since as UziTech says in MySQL databases are schemas, hmmm, this brings some complications... As @joaoe said, please send a PR if you can, @UziTech, and thanks for the report!

Reply:
I added #11456 to try to fix this without just reverting it.